tracee-ebpf/test/integration/integration_test.go

```diff
@@ -0,0 +1,256 @@
+package integration
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	ps "github.com/mitchellh/go-ps"
+	"github.com/onsi/gomega/gexec"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stretchr/testify/assert"
+)
+
+type Config struct {
+	TraceeBinaryPath string `required:"true" envconfig:"trc_bin"`
+}
+
+func getTraceeBinaryPath(t *testing.T) string {
+	var c Config
+	err := envconfig.Process("trc", &c)
+	require.NoError(t, err)
+
+	if _, err := os.Stat(c.TraceeBinaryPath); os.IsNotExist(err) {
+		require.FailNow(t, "failed to find tracee binary", err)
+	}
+	return c.TraceeBinaryPath
+}
+
+// load tracee into memory with args
+func loadTracee(t *testing.T, traceeBinPath string, w io.Writer, done chan bool, args ...string) {
+	cmd := exec.Command(traceeBinPath, args...)
+	fmt.Println("running: ", cmd.String())
+
+	session, err := gexec.Start(cmd, w, w)
+	require.NoError(t, err)
+	<-done
+	session.Interrupt()
+}
+
+// get pid by process name
+func getPidByName(t *testing.T, name string) int {
+	processes, err := ps.Processes()
+	require.NoError(t, err)
+
+	for _, p := range processes {
+		if strings.Contains(p.Executable(), name) {
+			return p.Pid()
+		}
+	}
+	return -1
+}
+
+// small set of actions to trigger a magic write event
+func checkMagicwrite(t *testing.T, gotOutput *bytes.Buffer) {
+	// create a temp dir for testing
+	d, err := ioutil.TempDir("", "Test_MagicWrite-dir-*")
+	require.NoError(t, err)
+
+	// cp a file to trigger
+	f, err := os.CreateTemp(d, "Test_MagicWrite-file-*")
+	require.NoError(t, err)
+	defer func() {
+		os.Remove(d)
+	}()
+
+	f.WriteString(`foo.bar.baz`)
+	f.Close()
+
+	cpCmd := exec.Command("cp", f.Name(), filepath.Join(d+filepath.Base(f.Name())+"-new"))
+	fmt.Println("executing: ", cpCmd.String())
+	cpCmd.Stdout = os.Stdout
+	assert.NoError(t, cpCmd.Run())
+
+	// check tracee output
+	assert.Contains(t, gotOutput.String(), `[102 111 111 46 98 97 114 46 98 97 122]`)
+}
+
+// execute a ls command
+func checkExeccommand(t *testing.T, gotOutput *bytes.Buffer) {
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// check tracee output
+	processNames := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, pname := range processNames {
+		assert.Equal(t, "ls", pname)
+	}
+}
+
+// only capture new pids after tracee
+func checkPidnew(t *testing.T, gotOutput *bytes.Buffer) {
+	traceePid := getPidByName(t, "tracee")
+
+	// run a command
+	_, _ = exec.Command("ls").CombinedOutput()
+
+	// output should only have events with pids greater (newer) than tracee
+	pids := strings.Split(strings.TrimSpace(gotOutput.String()), "\n")
+	for _, p := range pids {
+		pid, _ := strconv.Atoi(p)
+		assert.Greater(t, pid, traceePid)
```

Comment:
Although rare, but this may fail if the pids in the system are wrapped.

Reply:
Oh I'm not familiar with the concept of pids being wrapped, could you explain a little more?

Reply:
Depends on how the system is configured, there is a max pid number, after which wrapping occur.
You can see the value by:
`cat /proc/sys/kernel/pid_max`
On 64 bit systems, this number will usually be 4194304

Reply:
> > I do agree with Itay on this handling some things that should be handled in Makefiles, but regardless I think this is a great step in the right direction for the project.
> > There's some fragility with the way you're checking output, this test fails on my system:
> > ```
> > CONT  Test_Events/trace_filesystem_events_from_comm_ls
> > running:  ../tracee-ebpf/dist/tracee-ebpf --trace s=fs --trace comm=ls --output gotemplate=eventName.tmpl
> >     integration_test.go:162:
> >         	Error Trace:	integration_test.go:162
> >         	            				integration_test.go:226
> >         	Error:      	[]string{"access", "openat", "fstat", "close", "read", "pread64", "statfs", "ioctl", "statx", "getdents64", "write"} does not contain "lgetxattr"
> >         	Test:       	Test_Events/trace_filesystem_events_from_comm_ls
> > ```
> > 
> > 
> > I suppose somewhere on my system (shell, kernel, or lib version) uses a different system call or something of that nature. This again relates to our need for infrastructure where we can run github actions against lots of different kernel/distro versions. Just something to note for now.
> 
> Thanks this is a good point. Yeah actually this assertion is incomplete. There are many more sys calls that belong to the `fs` set.
> 
> https://github.com/aquasecurity/tracee/blob/228c6d329e86dd8f62bb5fe48761e78b67dafb39/tracee-ebpf/tracee/consts.go#L160-L528
> 
> that we will need to cover. I will find a better way to handle this.
> > At the end of the day I think integration tests matter most for tracee-rules. It doesn't really matter if the output events contain `execve` or `execveat`, as long as detections are working properly.
> 
> I agree but as we discussed on a call in the past, I think it ultimately depends on what our user interface is going to be for people to interface with tracee. Some might want just use it as a detection tool like you mentioned in which case adding integration tests for tracee-rules is the right place to start. Until we can finalize, better to have some coverage in the meantime.

I added a helper func here: https://github.com/aquasecurity/tracee/pull/606/commits/2dabbfa227c78f23d5224f2a100c269144617748#diff-02603f54659cbb5a36b20ea4235fc2b31a45e179c0e3ebdbfe84822b2403d81eR205-R214